src/NServiceBus.Core/Pipeline/Incoming/PendingTransportOperations.cs

```diff
@@ -1,22 +1,25 @@
 namespace NServiceBus
 {
+    using System.Collections.Concurrent;
     using System.Collections.Generic;
+    using System.Linq;
     using Transports;
 
     class PendingTransportOperations
-    {  
-        public IReadOnlyCollection<TransportOperation> Operations => operations;
+    {
+        public IReadOnlyCollection<TransportOperation> Operations => new List<TransportOperation>(operations);
```

Comment:
Concurrent* does this internally when you GetEnumerable() anyhow. Besides, Concurrent* doesn't implement IReadOnlyCollection in 4.5.2.